docs/develop/ethereum-to-ton/tvm-vs-evm.md

```diff
@@ -0,0 +1,86 @@
+# TVM vs EVM
+
+Ethereum Virtual Machine (EVM) and TON Virtual Machine (TVM) are both stack-based virtual machines developed for running smart contract code. Although they have common features, there are notable distinctions between them.
+
+## Data presentation
+
+### Ethereum Virtual Machine (EVM)
+1. Fundamental Data Units
+ - The EVM operates primarily on 256-bit integers, reflecting its design around Ethereum's cryptographic functions (e.g.,   Keccak-256 hashing and elliptic curve operations).
+ - Data types are limited mainly to integers, bytes, and occasionally arrays of these types, but all must conform to 256-bit processing rules.
+2. State Storage
+- The entire state of the Ethereum blockchain is a mapping of 256-bit addresses to 256-bit values. This mapping is maintained in a data structure known as the Merkle Patricia Trie (MPT).
+- The MPT enables Ethereum to efficiently prove the consistency and integrity of the blockchain state through cryptographic verification, which is vital for a decentralized system like Ethereum.
+3. Data Structure Limitations
+- The simplification to 256-bit word constraints means that the EVM is not inherently designed to handle complex or custom data structures directly.
+- Developers often need to implement additional logic within smart contracts to simulate more complex data structures, which can lead to increased gas costs and complexity.
+
+### TON Virtual Machine (TVM)
+1. Cell-Based Architecture
+- TVM uses a unique "bag of cells" model to represent data. Each cell can contain up to 128 data bytes and can have up to 4 references to other cells.
+- This structure allows the TVM to natively support arbitrary algebraic data types and more complex constructions such as trees or directed acyclic graphs (DAGs) directly within its storage model.
+2. Flexibility and Efficiency
+- The cell model provides significant flexibility, enabling the TVM to handle a wide variety of data structures more naturally and efficiently than the EVM.
```

Comment:
Didn’t get the point. It would be nice to have some real-life examples

Reply:
There is an example in the next paragraph. Maybe I misunderstood what you wanted to see.